include/helper.h

```diff
@@ -58,6 +58,21 @@ float interpolate(const float a[], const float b[], size_t size, float value_a);
  */
 const char *byte2bitstr(uint8_t b);
 
+/**
+ * @brief Check if a buffer is entirely zeroed.
+ *
+ * This function checks if all elements in a buffer are zero. For buffers equal to or larger than
+ * the size of `size_t`, it performs an optimized check by casting the buffer to a `size_t` pointer
+ * and examining the first `size_t`-sized block. If this block is zero, it then utilizes memcmp to
+ * compare the remaining part of the buffer against zero. For buffers smaller than `size_t`,
+ * the function conducts a byte-wise comparison.
```

Comment:
I'd suggest to move the explanation of the implementation details to the function itself (into the C file), as the user of the function probably doesn't care about how it works, just that it works.